### ISSUE TITLE
Fix JSON field naming

### ISSUE THREAD

**Issue by tbkka** [apple/swift-protobuf#82]
Today, there is a `toJsonFieldName()` function used in `MessageFieldGenerator.swift` to construct the JSON name for a field, instead of using the name provided by protoc in `descriptor.jsonName`.

There were originally two reasons for this:
* It was an attempt to try to pass Google's conformance tests.  I believe protoc still populates `descriptor.jsonName` in a way that does not pass the conformance tests.  Until that is fixed, it's not clear what the correct JSON name should be.  (Though I was told at one time that the conformance test was right and protoc was wrong.)
* It was an attempt to see if we could support JSON when used with protoc 2.6.

The right answer is probably to just stop trying to be clever:
* Remove `toJsonFieldName()`
* Use `descriptor.jsonName` and wait for either protoc or the conformance test to get fixed
* Don't try to support JSON for proto2 or when using protoc 2.6.  Work to split out the JSON generation should make this easier.

**Comment by ShaneQi** [comment]
How about `json_name` option?

```
message Foo {
    string bar = 1 [json_name="my_json_field_name"];
}
```

- In our [document](https://github.com/apple/swift-protobuf/blob/master/Documentation/PLUGIN.md), it says:  
  __In particular, note that the "my_my" field name in the proto file gets translated to "myMy" in the JSON serialized form. You can override this with a json_name property on fields as needed.__
- I tried and it turned out `json_name` option doesn't work in swift-protobuf (works in protobuf-java).

Does it because I used `json_name` option in wrong way, or just the documenting mistake?

**Comment by thomasvl** [comment]
Odds are it is because the data is currently ignored in the generator.

**Comment by tbkka** [comment]
Yep.  You're right.  That is the reason we must use `descriptor.jsonName` for the JSON name when it exists.

**Comment by ShaneQi** [comment]
You guys are right. I replaced `toJsonFieldName(descriptor.name!)` with `descriptor.jsonName!`.

Then `[json_name="my_json_field_name"]` started working.

I think removing `toJsonFieldName()` and using `descriptor.jsonName` is not a bad idea. Conformance test should be and will be fixed.

**Comment by tbkka** [comment]
This is fixed in pr #95.
